docs/proposals/choosing-scheduler.md

```diff
@@ -0,0 +1,165 @@
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Choosing scheduler in a multi-scheduler configuration
+
+## Introduction
+
+This document proposes a mechanism for assigning a scheduler to a pod
+in a system which is using multiple schedulers as described [here](multiple-schedulers.md).
+In particular, we decribe an admission controller which adds the scheduler name annotation,
+and which takes its configuration from a new API object called `PodPolicy` that determines,
+for each pod, which scheduler is responsible for scheduling that pod, based on various
+characteristics of the pod.
+
+## Implementation
+
+We will create a new admission controller called PodSchedulerPolicy. It will be configured
+using a new API object called `PodPolicy`. We have chosen a generic name and API
+for this object because we want to use the same object for defining other policies based on pod
+characteristics, for example see #17097.
+
+PodSchedulerPolicy will iterate through the `Rules` of the `PodPolicy` and apply the
+first one that matches the pod. Each `Rule` is a string specifying the scheduler name
+to apply to the pod if the pod satisfies the rule's PolicyPredicate, which currently is
+just a wrapper over `PodSelector`.
+
+## API
+
+### PodPolicy
+
+```go
+// A PodPolicy is a list of rules that define the policy. Each rule is tested
+// in order, and the first one that matches is applied.
+type PodPolicy struct {
+	Rules []PodPolicyRule `json:"rules,omitempty"`
+}
+
+type PodPolicyRule struct {
+	// The rule defined by Policy is applied if the pod satisfies all of the criteria in PolicyPredicate
+	// (currently this just means satisfying the PodSelector in PolicyPredicate).
+	// The reason we use PolicyPredicate instead of just putting *PodSelector here is to allow
+	// us to cleanly add additional predicates in the future; they would be added to
+	// PolicyPredicate, allowing PodPolicyRule to stay simple.
+	PolicyPredicate *PolicyPredicate `json:"policyPredicate,omitempty"`
+	Policy string `json:"policy"`
+}
+
+// Note that Namespace is not part of PolicyPredicate because we assume pod policies are per-Namespace.
+type PolicyPredicate struct {
+	PodSelector *PodSelector `json:"podSelector,omitempty"`
+}
+```
+
+### PodSchedulerPolicy admission controller
+
+```go
+// PodSchedulerPolicySpec defines the policy configuration for the PodSchedulerPolicy admission controller.
+type PodSchedulerPolicySpec struct {
+	Policy PodPolicy `json:"policy,omitempty"`
+}
+
+// PodSchedulerPolicyStatus just reflects back the policy set in the spec.
+type PodSchedulerPolicyStatus struct {
+	Policy PodPolicy `json:"policy,omitempty"`
+}
+
+// PodSchedulerPolicy describes the PodSchedulerPolicy admission controller,
+// which maps pods to the scheduler responsible for scheduling them.
+type PodSchedulerPolicy struct {
```

Comment:
No. The idea is that the admission controller resource is called PodSchedulerPolicy -- arguably a bad name, maybe PodSchedulerAssigner would be better -- and the PodPolicy is part of its Spec. The PodPolicy is intended to be generic and reusable for other components that want to assign policies based on matching on pod characteristics, while the PodSchedulerPolicy is an admission controller that uses a PodPolicy to assign pods to schedulers (by writing the scheduler name annotation on the pod).